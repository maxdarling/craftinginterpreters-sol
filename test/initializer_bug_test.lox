// Test to demonstrate the initializer return value bug

print "=== Testing Initializer Return Value Bug ===";

// This test should fail at compile time, but if it doesn't, it demonstrates the bug
class BuggyInit {
  init(value) {
    this.value = value;
    return "this should be ignored"; // This should cause a compile error
  }
}

// If the above compiles, this would demonstrate the bug:
// var buggy = BuggyInit(42);
// print buggy; // Should print "BuggyInit instance", not "this should be ignored"

print "Test completed - check if initializer return value causes compile error";
