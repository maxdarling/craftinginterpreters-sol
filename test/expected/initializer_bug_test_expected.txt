[line 9] Error at 'return': Can't return a value from an initializer.
